src/components/headless/components/data/item-scroller/transition.component.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from 'react';
+
+interface TransitionProps {
+  children: (state: string) => React.ReactNode;
+  in: boolean;
+  timeout: number;
+  onEnter?: () => void;
+  onEntering?: () => void;
+  onEntered?: () => void;
+  onExit?: () => void;
+  onExiting?: () => void;
+  onExited?: () => void;
+}
+
+function Transition({
```

Comment:
I prefer to use the `const` declaration format; for example `const Transition = ({ <props> }) => {` all other components in the library follow this format.